OpenSim/Actuators/VandenBogert2011Muscle.h

```diff
@@ -0,0 +1,568 @@
+                        \li 1: activation (unitless)*/
+        SimTK::Vec2 getResidual(const SimTK::State& s,
+                                double projFibVelNorm_guess,
+                                double activdot_guess, double u) const;
+
+
+        //Hacks because cache variables not implemented yet
+
+        /**@param The state of the system.
+           @returns The Fiber Length of the muscle.*/
+        double  getFiberLength(SimTK::State& s) const;
+        /**@param The state of the system.
+           @returns The activation of the muscle.*/
+        double  getActivation(SimTK::State& s) const;
+
+
+        //Temporary struct for troubleshooting
+        struct ImplicitResidual {
+            double forceResidual = SimTK::NaN;
+            double activResidual = SimTK::NaN;
+            double forceTendon = SimTK::NaN;
+            SimTK::Mat22 df_dy = {SimTK::NaN,SimTK::NaN,SimTK::NaN,
+                                  SimTK::NaN,SimTK::NaN,SimTK::NaN};
+            SimTK::Mat22 df_dydot = {SimTK::NaN,SimTK::NaN,SimTK::NaN,
+                                     SimTK::NaN,SimTK::NaN,SimTK::NaN};
+            SimTK::Vec2 df_du = {SimTK::NaN,SimTK::NaN};
+            double df_dmuscleLength = SimTK::NaN;
+            double F1 = SimTK::NaN;
+            double F2 = SimTK::NaN;
+            double F3 = SimTK::NaN;
+            double F4 = SimTK::NaN;
+            double F5 = SimTK::NaN; };
+
+//==============================================================================
+// MUSCLE.H INTERFACE
+//==============================================================================
+        /** @param[in] s The state of the system.
+            @returns The tensile force the muscle is generating (N). */
+        double computeActuation(const SimTK::State& s) const override final;
+
+        /** Computes the fiber length such that the fiber and tendon are
+        developing the same force, distributing the velocity of the entire
+        musculotendon actuator between the fiber and tendon according to their
+        relative stiffnesses.
+                @param[in,out] s The state of the system. */
+        void computeInitialFiberEquilibrium(SimTK::State& s) const override;
+
+        /** Computes the fiber length such that the fiber and tendon are
+        developing the same force, assuming velocities are zero. This is a
+        static equilibrium version of computeInitialFiberEquilibrium(). By
+        setting velocities to zero, we obtain a reasonable and robust solution
+        that provides a rough solution for fiber length.
+                @param[in,out] s The state of the system. */
+        void computeFiberEquilibriumAtZeroVelocity(SimTK::State& s) const
+                override;
+
+
+//=============================================================================
+// COMPUTATION
+//=============================================================================
+
+    // TODO: These should move to protected at some point
```

Comment:
I agree; the methods returning `ImplicitResidual` should be protected. The helpers below that can be public.